sql/item_strfunc.cc

```diff
@@ -1302,6 +1302,133 @@ bool Item_func_replace::fix_length_and_dec()
 }
 
 
+bool Item_func_sformat::fix_length_and_dec()
+{
+  ulonglong char_length= 0;
+  
+  if (agg_arg_charsets_for_string_result(collation, args, arg_count))
+    return TRUE;
+  
+  for (uint i=0 ; i < arg_count ; i++)
+    char_length+= args[i]->max_char_length() * 10; // temporal solution
+  
+  fix_char_length_ulonglong(char_length);
+  return FALSE;
+}
+
+longlong Item_func_sformat::locate(String *a, String *b, longlong offset)
+{
+  /* must be longlong to avoid truncation */
+  longlong   start=  0;
+  my_match_t match;
+  
+  if (offset > 0)
+  {
+    start= offset;
+    
+    if ((start <= 0) || (start > a->length()))
+      return 0;
+    offset--;
+    start--;
+    
```

Comment:
I going to leave it like this for now, because it breaks the logic of substr. I tried to fix it changing a few lines, but I couldn't, so I think it needs more analysis.